ql/src/java/org/apache/hadoop/hive/ql/io/arrow/Deserializer.java

```diff
@@ -369,7 +387,7 @@ private void readList(FieldVector arrowVector, ListColumnVector hiveVector, List
 
     read(arrowVector.getChildrenFromFields().get(0),
         hiveVector.child,
-        typeInfo.getListElementTypeInfo());
+        typeInfo.getListElementTypeInfo(), false);
```

Comment:
I encoded non-nested string vectors only, because it was hard to encode nested string vector. I originally assumed that Hive and Arrow vectors use same types except minor differences. However, a dictionary encoding results a new vector in a different type with a dictionary. So it brought bigger code changes.

I changed Serializer to encode nested string vectors in the second commit. All nested string vectors will be encoded now. Thanks for good feedback!